### PR TITLE
[SPARK-14917][SQL] Enable some ORC compressions tests for writing

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

https://issues.apache.org/jira/browse/SPARK-14917

As it is described in the JIRA, it seems Hive 1.2.1 which Spark uses now supports snappy and none.

So, this PR enables some tests for writing ORC files with compression codes, `SNAPPY` and `NONE`.
## How was this patch tested?

Unittests in `OrcQuerySuite` and `sbt scalastyle`.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
BTW, in `OrcHadoopHsRelationSuite` has a little bit inappropriate test because the default codec is `ZLIB` but the test `SPARK-13543: Support for specifying compression codec for ORC via option()` tries to test `compression` option by setting the value as `ZlIb`. This does not test if the compression is properly set up or not.

This is being handled in https://github.com/apache/spark/pull/12629 but not sure if this is merged (or reviewed). I can add this change in this PR.

**Comment by HyukjinKwon** [comment]
Could I maybe cc @srowen please?

**Comment by srowen** [comment]
I'm not familiar with this part of the code

**Comment by HyukjinKwon** [comment]
@rxin Could you please take a look?

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala [hunk 1]
```diff
@@ -169,39 +169,42 @@ class OrcQuerySuite extends QueryTest with BeforeAndAfterAll with OrcTest {
...
     val data = (1 to 100).map(i => (i, s"val_$i"))
     val conf = sqlContext.sessionState.hadoopConf
 
+    withOrcFile(data) { file =>
+      val expectedCompressionKind =
+        OrcFileOperator.getFileReader(file).get.getCompression
+      assert(CompressionKind.ZLIB === expectedCompressionKind)
+    }
+
     conf.set(ConfVars.HIVE_ORC_DEFAULT_COMPRESS.varname, "SNAPPY")
```
Comment (yhuai) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: Can you set this in the option in DataFrameWriter? It should work now (we propagate the conf to the underlying hadoop conf when creating the writer).

Also, let's use the string format of the key directly.

Comment (HyukjinKwon) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcQuerySuite.scala]: Sure! Thank you.

**Comment by yhuai** [comment]
Cool. Thanks! lgtm

**Comment by HyukjinKwon** [comment]
Thank you @yhuai! 
Actually, do you mind if I try to submit a PR after walking through sql/hive tests and remove the class imports where it is possible maybe? I won't if there is any possible problem with that (eg. difficult to review or easy to make a conflict..)

**Comment by yhuai** [comment]
How about we do it a little bit later? Maybe that will introduce conflicts  with prs from others.

**Comment by HyukjinKwon** [comment]
@yhuai Sure. Thank you.

**Pull request closed without merging**